internal/cmdsearchreplace/cmdsearchreplace_test.go

```diff
@@ -0,0 +1,453 @@
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/cmd/config/ext"
+	"sigs.k8s.io/kustomize/kyaml/copyutil"
+)
+
+func TestSearchCommand(t *testing.T) {
+	var tests = []struct {
+		name              string
+		input             string
+		args              []string
+		out               string
+		expectedResources string
+		errMsg            string
+	}{
+		{
+			name: "search replace by value",
+			args: []string{"--by-value", "3", "--put-literal", "4"},
+			out: `${baseDir}/
+matched 2 field(s)
+${filePath}:  spec.replicas: 4
+${filePath}:  foo.bar: 4
+`,
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+---
+apiVersion: apps/v1
+kind: Service
+metadata:
+  name: nginx-service
+foo:
+  bar: 3
+ `,
+			expectedResources: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 4
+---
+apiVersion: apps/v1
+kind: Service
+metadata:
+  name: nginx-service
+foo:
+  bar: 4
+ `,
+		},
+		{
+			name: "search by value",
+			args: []string{"--by-value", "3"},
+			out: `${baseDir}/
+matched 2 field(s)
+${filePath}:  spec.replicas: 3
+${filePath}:  spec.foo: 3
+`,
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+  foo: 3
+---
+apiVersion: apps/v1
+kind: Service
+metadata:
+  name: nginx-service
+ `,
+			expectedResources: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+  foo: 3
+---
+apiVersion: apps/v1
+kind: Service
+metadata:
+  name: nginx-service
+ `,
+		},
+		{
+			name: "search replace by regex",
+			args: []string{"--by-value-regex", "nginx-*", "--put-literal", "ubuntu-deployment"},
+			out: `${baseDir}/
+matched 1 field(s)
+${filePath}:  metadata.name: ubuntu-deployment
+`,
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+ `,
+			expectedResources: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ubuntu-deployment
+spec:
+  replicas: 3
+ `,
+		},
+		{
+			name: "search by regex",
+			args: []string{"--by-value-regex", "nginx-*"},
+			out: `${baseDir}/
+matched 1 field(s)
+${filePath}:  metadata.name: nginx-deployment
+`,
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+ `,
+			expectedResources: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+ `,
+		},
+		{
+			name: "put literal error",
+			args: []string{"--put-literal", "something"},
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+ `,
+			expectedResources: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+ `,
+			errMsg: `at least one of ["by-value", "by-value-regex", "by-path"] must be provided`,
+		},
+		{
+			name: "add non-existing field",
```

Comment:
Add a test for changing fields that don't respect the schema even if it shows the incorrect behavior. In the future, when OpenAPI is referenced, you can just flip the expected behavior.

Reply:
how about a test for incorrect types (e.g. put a string, where a number is expected) => Isn't this same as the comment above ?

Reply:
Tracking this in another comment.